### ISSUE TITLE
Document User location functionality

### ISSUE THREAD

**Issue by aaronschachter** [DoSomething/product#44]
UPDATED ISSUE:
Document how we capture and save User location info

Original post:

Update designs to remove Location from Hub, we don't capture this information on the web or in mobile.

https://github.com/DoSomething/product/blob/master/LetsDoThis/Screens/Hub/Public%20Hub*.png

**Comment by marahml** [comment]
Can't we detect users' locations & use if we ask permission? Location was a big part of the discussion with mgmt. cc @jonuy

**Comment by aaronschachter** [comment]
I was not aware that we are doing this.  Is this implemented in the Android app?

There are a few potential issues here: 
- If I'm on vacation in Alaska and I download and install the app, it will save my location as Alaska to my profile
- Complexity around translating the longitude/latitude coordinates from phone to city name
- Using Location will require more battery usage

**Comment by marahml** [comment]
Yes, this has always been the plan. I'm waiting to see the updated Android app once @jonuy finishes bug fixes so it will stop crashing - but it should be in there.

In terms of implementation, the good news is that lots of other folks have taken advantage of this native functionality before and we can learn from them. 

Here are some details on how to implement from apple: https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/LocationAwarenessPG/CoreLocation/CoreLocation.html

I'm less worried about battery issues. Also, on iOS users have the ability to turn off location detection - and I'd assume it'd be similar for android as well.

**Comment by aaronschachter** [comment]
@marahml I'd imagine we need the ability for a User to edit their location. So if I download the app while I'm on vacation in Alaska, my profile is not forever set to Alaska. Perhaps not needed for launch, but we should document whatever decision we land on for v1.0

**Comment by marahml** [comment]
@aaronschachter would the "share my location" permission allow us to just check every time a user opens the app?

**Comment by aaronschachter** [comment]
Yes, but we would still want to save the location to the User document in Northstar in order to display it from the Hub (if someone else views their profile). Let's chat through this during standup today. cc @jonuy

**Comment by marahml** [comment]
Hub: Save user location based on where they open the app the first time. 
Reportback: Display user location based on location saved to profile

Ask for permission to use user's location
- Possible that we will get users without location (need designs for gallery w/ and w/o location) 
- In V2 allow them to change it in settings.

**Comment by aaronschachter** [comment]
Documenting our followup convo in Slack:

Priority is posting upon new user registration. Nice to have is posting upon login if we don't have a location set for the User already.

**Comment by aaronschachter** [comment]
Noting that we will only store the User's `country` . cc @marahml @jonuy @lkpttn

**Comment by aaronschachter** [comment]
Closing this out: we'll store the User's country to the `country_code` property if access to the phone location is granted.

**Comment by aaronschachter** [comment]
Actually re-open until we document the README.
